assets/js/modules/adsense/components/dashboard/DashboardTopEarningPagesWidget.js

```diff
@@ -0,0 +1,140 @@
+/**
+ * DashboardTopEarningPagesWidget component.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { __, _x } from '@wordpress/i18n';
+import { compose } from '@wordpress/compose';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { STORE_NAME as ANALYTICS_STORE } from '../../../analytics/datastore/constants';
+import { STORE_NAME as CORE_USER } from '../../../../googlesitekit/datastore/user/constants';
+import whenActive from '../../../../util/when-active';
+import PreviewTable from '../../../../components/preview-table';
+import { getDataTableFromData, TableOverflowContainer } from '../../../../components/data-table';
+import Layout from '../../../../components/layout/layout';
+import AdSenseLinkCTA from '../../../analytics/components/common/AdSenseLinkCTA';
+import getDataErrorComponent from '../../../../components/notifications/data-error';
+import getNoDataComponent from '../../../../components/notifications/nodata';
+
+const { useSelect } = Data;
+
+function DashboardTopEarningPagesWidget() {
+	const {
+		data,
+		error,
+		loading,
+	} = useSelect( ( select ) => {
+		const store = select( ANALYTICS_STORE );
+		const args = {
+			dateRange: select( CORE_USER ).getDateRange(),
+			dimensions: [ 'ga:pageTitle', 'ga:pagePath' ],
+			metrics: [
+				{ expression: 'ga:adsenseRevenue', alias: 'Earnings' },
+				{ expression: 'ga:adsenseECPM', alias: 'Page RPM' },
+				{ expression: 'ga:adsensePageImpressions', alias: 'Impressions' },
+			],
+			orderby: {
+				fieldName: 'ga:adsenseRevenue',
+				sortOrder: 'DESCENDING',
+			},
+			limit: 10,
+		};
+
+		return {
+			data: store.getReport( args ),
+			error: store.getErrorForSelector( 'getReport', [ args ] ),
+			loading: store.isResolving( 'getReport', [ args ] ),
+		};
+	} );
+
+	if ( loading ) {
+		return (
+			<PreviewTable rows={ 5 } padding />
+		);
+	}
+
+	if ( error ) {
+		// Specifically looking for string "badRequest"
+		if ( 'badRequest' === error?.data?.reason ) {
+			return (
+				<Layout className="googlesitekit-top-earnings-pages" fill>
+					<AdSenseLinkCTA />
+				</Layout>
+			);
+		}
+		return getDataErrorComponent( 'adsense', error.message, false, false, false, error );
+	}
+
+	if ( ! data || ! data[ 0 ]?.data?.rows ) {
```

Comment:
```suggestion
	if ( ! data || ! data.length || ! data[ 0 ]?.data?.rows ) {
```
To avoid invalid access of `data[ 0 ]` after.